shell/common/shell.h

```diff
@@ -101,7 +99,9 @@ class Shell {
   TracingController tracing_controller_;
 
   std::vector<ftl::WeakPtr<Rasterizer>> rasterizers_;
-  std::vector<ftl::WeakPtr<PlatformView>> platform_views_;
+  std::vector<std::weak_ptr<PlatformView>> platform_views_;
+
+  std::recursive_mutex mutex_;
```

Comment:
It is needed otherwise the lock inside `WaitForPlatformViewIds` will acquire it and the nested invocation of `GetPlatformViews` will freeze.